core/include/gnuradio-4.0/BlockRegistry.hpp

```diff
@@ -26,44 +26,53 @@ std::unique_ptr<gr::BlockModel> blockFactory(property_map params) {
 } // namespace detail
 
 class BlockRegistry {
-    using TBlockTypeHandler = std::function<std::unique_ptr<gr::BlockModel>(property_map)>;
+    struct TBlockTypeHandler {
+        std::string                                                  alias;
+        std::function<std::unique_ptr<gr::BlockModel>(property_map)> createFunction;
+    };
 
-    std::vector<std::string_view> _blockTypes;
-
-    std::unordered_map<std::string_view, std::unordered_map<std::string_view, TBlockTypeHandler>> _blockTypeHandlers;
-
-    auto& findBlockTypeHandlersMap(std::string_view blockType) {
-        if (auto it = _blockTypeHandlers.find(blockType); it != _blockTypeHandlers.end()) {
-            return it->second;
-        }
-        _blockTypes.emplace_back(blockType);
-        return _blockTypeHandlers[blockType];
-    }
+    std::vector<std::string>                              _blockTypes;
+    std::map<std::string, TBlockTypeHandler, std::less<>> _blockTypeHandlers;
```

Comment:
This is basically a `std::map<string, pair<string, func>>`. Why do we need both strings (key and it's alias)?

If needed, maybe add a small comment on the layout and few examples that developers should expect to find in this map.

Reply:
When writing out the block type (GRC/message), the meta::type_name on the block returns the long name (e.g. `SomeBlockImpl<T, true, 42>`>) and `BlockRegistry::blockTypeName("SomeBlockImpl<T, true, 42>")` does a lookup in this map and returns the alias if set (e.g. `SomeBlock<T>`). This could also be a separate `map<string, string>`, given that types with alias are the exception.